Fix & clean up medical conditions

screret opened the pull request:
## What
- Fixed medical conditions caused by holding hazardous materials (e.g. cancer from uranium) not updating correctly.
- Added some much needed documentation to the system and made the `/medical_condition query` command's output nicer to read.
- Rebalanced the medical conditions so their symptoms take longer to apply and aren't as drastic

I also chose to remove berylliosis and silicosis entirely, as they didn't actually do much and were just annoying to have around.

## Implementation Details
- MADE MEDICAL CONDITIONS USE AN ACTUAL REGISTRY INSTEAD OF A MAP
  - this shouldn't affect anyone too much, as I don't know anyone who actually used the system
- Fixed invulnerability/hitstun/etc. breaking the death symptom
- Fixed high levels of resistance being able to block medical condition damage (like death.)
- Fixed the player's max health not updating until they take damage
- Fixed the player's max breath (the air bubbles) not updating at all
- Made the hazardous item tracker respect the held items' stack sizes and material amounts where they previvously didn't.
- Renamed some methods' parameters to make more sense and added clarifying comments to them & others
- Added separate lang keys for conditions affecting a player so the output of `/medical_condition query` makes sense. It now display something like this ([1](#new-display)) instead of the previous ([2](#old-display))
### new display
> Player Dev has
Cancer for 10 minutes 5 seconds

### old display
> Player Dev has these medical conditions:
Condition Carcinogenic: 10 minutes 5 seconds

## Outcome
- Players' health and air is actually displayed as going down
- Players get the intermediate symptoms instead of, say, randomly dying from cancer
- People won't get blinded by methanol the instant they touch it
- Cancer now takes **much** longer to do anything drastic, and after that, the effects take longer to reach their full potential

## How Was This Tested
_I wrote actual tests for a feature!_ (for once, lol)

I also tested this in game quite extensively, mostly by running `/medical_condition apply @s gtceu:carcinogen <amount>` over and over with different numbers in the 1-36000 range and observing what attributes changed & by how much with `/attribute @s <attribute id> get`.
I also tested some other conditions such as arsenicosis and CO poisoning, they also worked correctly.

Same was done for holding items by having a stack plutonium blocks in my inventory and tracking the cancer point gain with `/medical_condition query` and current symptoms with `/medical_condition query symptoms`.

## Additional Information
Balance people should look over the tweaks at least a bit

## Potential Compatibility Issues
The change to a "real" registry could make some people have issues but, as already mentioned, I don't know anyone who used the system.

screret committed:
clean up medical condition-related translation keys

screret committed:
make medical conditions use an actual registry instead of faking one

screret committed:
rename a couple methods' parameters, add clarifying comments to some, shortcuts for others

screret committed:
add documentation on how lang keys are chosen for medical condition

screret committed:
special case the health debuff condition so it can resync the player's health

screret committed:
add a "consumption" hazard trigger that triggers when a player eats something

screret committed:
symptom that gives the darkness effect (the one wardens use)

screret committed:
add separate "affected" text for most conditions because "Player <player> has: Carcinogenic" looks bad
fixes GregTechCEu/GregTech-Modern#2780

screret committed:
fix compile errors

screret committed:
spotless, datagen

screret committed:
make fastutil be loaded from maven central so we can get its source artifacts

screret committed:
merge IMedicalConditionTracker into MedicalConditionTracker
most uses required the concrete class anyway

screret committed:
use absolute values for symptoms for clarity

screret committed:
remove silicosis and berylliosis as they were only annoying and added even less than the other conditions

screret committed:
track the max air symptom's value changes in the entity rather than the tracker for efficiency

screret committed:
use fastutil reference maps for the progress tracking so we don't waste time comparing

screret committed:
add descriptions of the various "generic" conditions to GTMedicalConditions

screret committed:
spotless

screret committed:
tweak the language of `/medical_condition query` to be easier to read, datagen

screret committed:
respect the item's material amount when progressing conditions

screret committed:
reset air supply debuff when clearing effects

screret committed:
properly remove cleared conditions' symptoms

screret committed:
allow passing negative values into `/medical_condition apply <condition> [progression]`

screret committed:
tag all medical condition damage types as bypasses_cooldown

screret committed:
datagen

screret committed:
run datagen and fix the other translations

screret committed:
Merge branch '1.20.1' into sc/fix-medical-conditions

# Conflicts:
#	src/main/resources/assets/gtceu/lang/uk_ua.json

Reabstraction commented on gradle/scripts/repositories.gradle:
```diff
@@ -1,6 +1,11 @@
 repositories {
     mavenLocal()
     mavenCentral()
+    // force gradle to download FastUtil from the central maven so we can get source artifacts
```
This should be made into its own PR

screret committed:
Merge branch '1.20.1' into sc/fix-medical-conditions

# Conflicts:
#	src/generated/resources/assets/gtceu/lang/en_ud.json
#	src/generated/resources/assets/gtceu/lang/en_us.json
#	src/main/java/com/gregtechceu/gtceu/data/lang/LangHandler.java
#	src/main/resources/assets/gtceu/lang/uk_ua.json

screret committed:
Merge branch '1.20.1' into sc/fix-medical-conditions

# Conflicts:
#	src/main/resources/assets/gtceu/lang/uk_ua.json

screret committed:
more test util functions

screret committed:
Medical condition/hazard tests

screret committed:
oh I have to tick the player myself ok

screret committed:
some docs

screret committed:
add a mixin to make test method names be displayed in snake_case even if the actual method name is in camelCase

screret committed:
make antidote tooltips use the correct translation key for curable conditions

screret committed:
fix bad PPE tooltips

screret committed:
fix timeouts being too short

screret committed:
minor reordering

screret committed:
less cancer

screret committed:
use correct personal pronoun for Players

screret committed:
test the different hazard protection types

screret committed:
actually eat all the pills

screret committed:
rename event listener to better describe its function

screret committed:
use a forge FakePlayer so event listeners that require a ServerPlayer work properly

screret committed:
less fake player (one that actually ticks)

screret committed:
use the tick method that actually runs all the stuff we need

screret committed:
update the held item if need be

screret committed:
wrong value

screret committed:
rename method

screret committed:
lessen visibility because these don't need to be protected

screret committed:
fix test_medical_condition_ticking not ticking right

screret committed:
clean up `getAndAssertAttributeModifier`

screret committed:
im giving up on this damn test why it no work

screret committed:
Merge branch '1.20.1' into sc/fix-medical-conditions

# Conflicts:
#	gradle/scripts/repositories.gradle

screret committed:
fix compilation error

screret committed:
Merge branch '1.20.1' into sc/fix-medical-conditions

screret committed:
spleets

screret committed:
Merge branch '1.20.1' into sc/fix-medical-conditions

screret commented:
I switched the base branch to 1.20.1-v8.0.0 because of the retarget and then discovered that the pipe model rework hasn't been merged to that, so I undid the base branch change.
Turns out doing that dismisses all reviews.

screret committed:
implement toString for MedicalCondition and Symptom

screret committed:
make the `/medical_condition` command's output better

screret committed:
tick players' condition trackers based on the player's tick count instead of global game time
(spreads out the updating over multiple ticks instead of everything happening on the dot)

screret committed:
datagen, spotless

screret committed:
don't send query results to all OPs

screret committed:
fix mining fatigue symptom being way too harsh (and not actually lowering mining speed)

screret committed:
clean up this break speed event handler so it won't give 5x mining speed if underwater with an aqua affinity helmet

screret committed:
also move it to above the massive remap event listener

screret committed:
fix max slowness symptom deleting movement speed because the assumed default value is 7x too high

screret committed:
remove ConfiguredSymptom constructor that allowed only defining an absolute minimum threshold as it made the code a lot more convoluted

screret committed:
do the same for Symptom and also make some of the constructor variants static factory methods instead so what they're for is slightly clearer

screret committed:
fix some commands having very high permission requirements

screret committed:
spot less

screret committed:
Merge branch '1.20.1' into sc/fix-medical-conditions

# Conflicts:
#	src/main/java/com/gregtechceu/gtceu/api/capability/GTCapabilityHelper.java
#	src/main/java/com/gregtechceu/gtceu/api/capability/forge/GTCapability.java
#	src/test/java/com/gregtechceu/gtceu/gametest/util/TestUtils.java

screret committed:
Merge branch '1.20.1' into sc/fix-medical-conditions

# Conflicts:
#	src/main/java/com/gregtechceu/gtceu/api/data/chemical/material/properties/HazardProperty.java
#	src/main/java/com/gregtechceu/gtceu/common/CommonEventListener.java
#	src/main/java/com/gregtechceu/gtceu/common/machine/electric/AirScrubberMachine.java
#	src/test/java/com/gregtechceu/gtceu/gametest/util/TestUtils.java

screret committed:
Merge branch 'refs/heads/1.20.1' into sc/fix-medical-conditions

# Conflicts:
#	src/generated/resources/assets/gtceu/lang/en_us.json

screret committed:
fix compile errors

screret committed:
fix my tests

screret committed:
spotesss

screret committed:
ooops forgot to commit this
